File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/menus/OnlineSettingsFragment.java
package de.dimedis.mobileentry.fragments.menus;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

import de.dimedis.mobileentry.databinding.FragmentOnlinesettingsBinding;

public class OnlineSettingsFragment extends AbsSettingsMenuFragment {

    FragmentOnlinesettingsBinding binding;

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        public Builder() {
        }

        public OnlineSettingsFragment build() {
            return new OnlineSettingsFragment();
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        binding = FragmentOnlinesettingsBinding.inflate(inflater, container, false);
        binding.ok.setOnClickListener(view -> onClickOK());
        binding.cancel.setOnClickListener(view -> onClickCancel());

        return binding.getRoot();
    }

    @Override
    protected void init() {
        super.init();
        binding.switch1.setChecked(mSettings.isLocalScan());
        binding.switch1.setOnCheckedChangeListener((buttonView, isChecked) -> mSettings.setLocalScan(isChecked));
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/menus/OnlineSettingsFragment.kt
package de.dimedis.mobileentry.fragments.menus

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import de.dimedis.mobileentry.databinding.FragmentOnlinesettingsBinding

class OnlineSettingsFragment : AbsSettingsMenuFragment() {
    var binding: FragmentOnlinesettingsBinding? = null

    class Builder {
        fun build(): OnlineSettingsFragment {
            return OnlineSettingsFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentOnlinesettingsBinding.inflate(inflater, container, false)
        binding!!.ok.setOnClickListener {
            onClickOK() }
        binding!!.cancel.setOnClickListener {
            onClickCancel() }
        return binding!!.root
    }

    override fun init() {
        super.init()
        binding!!.switch1.isChecked = mSettings!!.localScan
        binding!!.switch1.setOnCheckedChangeListener { buttonView, isChecked ->
            mSettings!!.localScan = (isChecked) }
    }

    companion object {
        fun builder(): Builder {
            return Builder()
        }
    }
}